GUI/src/gui_qt.py

```diff
@@ -0,0 +1,458 @@
+        self.stop_script = os.path.join(self.config_folder, "stop_simulation")
+        
+        if self.setup_files() and self.pull_images() and self.build_and_set_config():
+            self.start_button.setEnabled(True)
+        else:
+            self.start_button.setEnabled(False)
+            self.selected_folder = None
+            self.config_combo.setEnabled(True)
+
+    def select_file(self, title, files):
+        if len(files) == 1:
+            return files[0]
+        
+        dialog = FileSelectDialog(title, files, self)
+        dialog.exec()
+        return dialog.get_selected()
+
+    def get_selected_route(self):
+        try:
+            with open(self.docker_compose_file, 'r') as f:
+                content = yaml.safe_load(f)
+            services = content.get('services', {})
+            carla_integration = services.get('carma-carla-integration_1', {})
+            command = carla_integration.get('command', '')
+            
+            for cmd_part in command.split():
+                if cmd_part.startswith("selected_route:="):
+                    return cmd_part.split("selected_route:=", 1)[1].strip("'\"")
+            
+            self.log_message("Error: selected_route not found in docker-compose.yml", "error")
+            return None
+        except Exception as e:
+            self.log_message(f"Error reading docker-compose.yml: {e}", "error")
+            return None
+
+    def setup_files(self):
+        map_folder = os.path.join(self.config_folder, "MAP")
+        map_files = [f for f in os.listdir(map_folder) if os.path.isfile(os.path.join(map_folder, f))]
+        
+        if not map_files:
+            self.log_message("Error: No files found in MAP folder.", "error")
+            return False
+        
+        selected_map = self.select_file("Select Map File", map_files)
+        if not selected_map:
+            self.log_message("Error: No map file selected.", "error")
+            return False
+        
+        selected_route_base = self.get_selected_route()
+        if not selected_route_base:
+            return False
+        
+        route_folder = os.path.join(self.config_folder, "route_config")
+        selected_route_file = f"{selected_route_base}.csv"
+        selected_route_path = os.path.join(route_folder, selected_route_file)
+        
+        if not os.path.isfile(selected_route_path):
+            self.log_message(f"Error: Route file {selected_route_file} not found in {route_folder}.", "error")
+            return False
+        
+        try:
+            # Copy map
+            dest_maps_dir = "/opt/carma/maps"
+            os.makedirs(dest_maps_dir, exist_ok=True)
+            dest_map = os.path.join(dest_maps_dir, "vector_map.osm")
+            if os.path.exists(dest_map):
+                now = datetime.now().strftime("%m%d%Y%H%M%S")
+                backup = os.path.join(dest_maps_dir, f"vector_map_{now}.osm.backup")
```

Comment:
Not necessary for current MVP version, but would be good to actually compare here not just if `dest_map` exists, but if `dest_map` and the desired map are actually different. This could possibly end up creating a bunch of `vector_map_*.osm.backup` files that are identical copies of each other if this is called multiple times.